presto-hive/src/main/java/com/facebook/presto/hive/PrestoS3FileSystem.java

```diff
@@ -764,17 +764,21 @@ public int read(byte[] buffer, int offset, int length)
                 int bytesRead = retry()
                         .maxAttempts(maxAttempts)
                         .exponentialBackoff(new Duration(1, TimeUnit.SECONDS), maxBackoffTime, maxRetryTime, 2.0)
-                        .stopOn(InterruptedException.class, UnrecoverableS3OperationException.class)
+                        .stopOn(InterruptedException.class, UnrecoverableS3OperationException.class, AbortedException.class)
                         .onRetry(STATS::newReadRetry)
                         .run("readStream", () -> {
                             seekStream();
                             try {
                                 return in.read(buffer, offset, length);
                             }
-                            catch (Exception e) {
-                                STATS.newReadError(e);
+                            catch (AbortedException abortedException) {
```

Comment:
Fair enough. For now I only added aborted exception to `stopOn()`. `seekStream()` is only called from the same retry loop, so we don't need to do anything additional for that.